nflow-engine/src/main/java/io/nflow/engine/internal/guice/EngineModule.java

```diff
@@ -0,0 +1,176 @@
+    return engineConfiguration.nflowThreadFactory();
+  }
+
+  @Provides
+  @Inject
+  @Singleton
+  public WorkflowInstanceExecutor nflowExecutor(@NFlow ThreadFactory factory, Environment env) {
+    return engineConfiguration.nflowExecutor(factory, env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public ObjectMapper nflowObjectMapper() {
+    return engineConfiguration.nflowObjectMapper();
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public DataSource nflowDataSource(Environment env) {
+    return getDatabaseConfiguration(env).nflowDatasource(env, metricRegistry);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public DatabaseInitializer nflowDatabaseInitializer(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowDatabaseInitializer(dataSource, env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public JdbcTemplate nflowJdbcTemplate(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowJdbcTemplate(dataSource);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public NamedParameterJdbcTemplate nflowNamedParameterJdbcTemplate(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowNamedParameterJdbcTemplate(dataSource);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public TransactionTemplate nflowTransactionTemplate(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowTransactionTemplate(new DataSourceTransactionManager(dataSource));
+  }
+
+  @Provides
+  @Singleton
+  @Inject
+  public SQLVariants nflowSQLVariants(Environment env) {
+    return getDatabaseConfiguration(env).sqlVariants();
+  }
+
+  private DatabaseConfiguration getDatabaseConfiguration(Environment env) {
+    DatabaseConfiguration db;
+    String dbtype = env.getProperty("nflow.db.type", String.class);
+    switch (dbtype) {
+    case "h2":
+      db = new H2DatabaseConfiguration();
+      break;
+    case "mysql":
+      db = new MysqlDatabaseConfiguration();
+      break;
+    case "oracle":
+      db = new OracleDatabaseConfiguration();
+      break;
+    case "postgresql":
+      db = new PgDatabaseConfiguration();
+      break;
+    default:
+      throw new RuntimeException("Unknown DB");
+    }
+    return db;
+  }
+
+  class EngineInitModule extends AbstractModule {
+    @Override
+    protected void configure() {
+      requestInjection(this);
+    }
+
+    @Inject
+    void initPostConstruct(WorkflowDefinitionService workflowDefinitionService) throws Exception {
+      workflowDefinitionService.postProcessWorkflowDefinitions();
+    }
+
+    @Inject
+    void initLifeCycleAutoStart(WorkflowLifecycle lifecycle) {
+      if (lifecycle.isAutoStartup()) {
+        lifecycle.start();
+      }
+    }
+
+    @Inject
+    void initH2TcpServer(Environment env) throws Exception {
+      Server server = new H2DatabaseConfiguration().h2TcpServer(env);
```

Comment:
The spring code only starts the server if the profile is `h2` - this seems to start it unconditionally.